### DISCUSSION TITLE
TypeError: Assignment to lists with more than one item is not supported’

### DISCUSSION THREAD

**Discussion by lnrabb** [nextgenhealthcare/connect#6110] [Q&A] [answered]
I created a Message Builder transformer to add a prefix to the beginning of OBR.2.1. The issue is that there are two OBR segments in the message, so I am getting the error of "TypeError: Assignment to lists with more than one item is not supported’".  How would I add a prefix to each OBR.2.1 field? I've seen where you can use the following code to iterate through each of the segments, but I am not sure how to add the prefix using Javascript. 

for each (obr2 in msg['OBR']['OBR.2']) {
//add prefix to OBR.2.1
}
Please see Message Builder with prefix of ABC added to OBR.2.1 as an example: 

Message Segment: msg['OBR']['OBR.2']['OBR.2.1']
Mapping: "ABC"+msg['OBR']['OBR.2']['OBR.2.1'].toString()

**Comment by lnrabb** [comment]
It seems that most people use JS. How would I add a prefix in JS? I am not
finding a way to do that.

Thank you!

On Tue, Feb 20, 2024 at 4:23 PM pacmano1 ***@***.***> wrote:

> If using the message builder, you would use an iterator
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nextgenhealthcare/connect/discussions/6110#discussioncomment-8535399>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A6DN3RTZC6JMYUYBTYLBHI3YUUH35AVCNFSM6AAAAABDR3ZZWSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DKMZVGM4TS>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 

*Lindsay Rabb*

Support Specialist

Medicom

434 Fayetteville Street #1650 | Raleigh, NC 27601

Office: 844.686.9599 Direct: 910.242.4282

<http://www.medicom.us/>
<https://www.linkedin.com/company/medicom-technologies-inc./>
<https://twitter.com/MedicomTech>

Confidentiality Notice: This e-mail message, including any attachments, is
for the sole use of the intended recipient(s) and may contain confidential
and privileged information. Any unauthorized review, use, disclosure, or
distribution is prohibited. If you are not the intended recipient, please
contact the sender by reply e-mail and destroy all copies of the original
message.
 
 ----
Secured by Paubox - HITRUST CSF certified
https://www.paubox.com
----

**Reply by pacmano1** [reply]
Put a before and after message here so we are clear what you are asking.   Use code snippets please.

But there are probably hundreds of examples on the mirth forums on iterating over segments, did you check there first?

**Comment by Hueli** [comment, marked as answer]
Probably this is, what you are looking for:

 ```javascript
for each (obr in msg.OBR)
  obr['OBR.2']['OBR.2.1'] = "ABC" + obr['OBR.2']['OBR.2.1'].toString();
```

**Comment by lnrabb** [comment]
Thank you so much!

On Fri, Feb 23, 2024 at 9:44 AM Hueli ***@***.***> wrote:

> Probably this is, what you are looking for:
>
> for each (obr in msg.OBR)
>  obr['OBR.2']['OBR.2.1'] = "ABC" + obr['OBR.2']['OBR.2.1'].toString();
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nextgenhealthcare/connect/discussions/6110#discussioncomment-8569163>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A6DN3RUVINIS4XCE2QR76W3YVCTMPAVCNFSM6AAAAABDR3ZZWSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DKNRZGE3DG>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 

*Lindsay Rabb*

Support Specialist

Medicom

434 Fayetteville Street #1650 | Raleigh, NC 27601

Office: 844.686.9599 Direct: 910.242.4282

<http://www.medicom.us/>
<https://www.linkedin.com/company/medicom-technologies-inc./>
<https://twitter.com/MedicomTech>

Confidentiality Notice: This e-mail message, including any attachments, is
for the sole use of the intended recipient(s) and may contain confidential
and privileged information. Any unauthorized review, use, disclosure, or
distribution is prohibited. If you are not the intended recipient, please
contact the sender by reply e-mail and destroy all copies of the original
message.
 
 ----
Secured by Paubox - HITRUST CSF certified
https://www.paubox.com
----
